diffsims/utils/sim_utils.py

```diff
@@ -448,3 +448,289 @@ def uvtw_to_uvw(uvtw):
+    >>> wavelength = 2.51/1000 # Electron wavelength for 200 kV
+    >>> angle = sim_utils.get_holz_angle(wavelength, lattice_size)
+
+    """
+    k0 = 1.0 / electron_wavelength
+    kz = 1.0 / lattice_parameter
+    in_root = kz * ((2 * k0) - kz)
+    sin_angle = (in_root ** 0.5) / k0
+    angle = np.arcsin(sin_angle)
+    return angle
+
+
+def scattering_angle_to_lattice_parameter(electron_wavelength, angle):
+    """Convert scattering angle data to lattice parameter sizes.
+
+    Parameters
+    ----------
+    electron_wavelength : float
+        Wavelength of the electrons in the electron beam. In nm.
+        For 200 kV electrons: 0.00251 (nm)
+    angle : NumPy array
+        Scattering angle, in radians.
+
+    Returns
+    -------
+    lattice_parameter : NumPy array
+        Lattice parameter, in nanometers
+
+    Examples
+    --------
+    >>> import diffsims.utils.sim_utils as sim_utils
+    >>> angle_list = [0.1, 0.1, 0.1, 0.1] # in radians
+    >>> wavelength = 2.51/1000 # Electron wavelength for 200 kV
+    >>> lattice_size = sim_utils.scattering_angle_to_lattice_parameter(
+    ...     wavelength, angle_list)
+
+    """
+
+    k0 = 1.0 / electron_wavelength
+    kz = k0 - (k0 * ((1 - (np.sin(angle) ** 2)) ** 0.5))
+    return 1 / kz
+
+
+def bst_to_beta(bst, acceleration_voltage):
+    """Calculate beam deflection (beta) values from Bs * t.
+
+    Parameters
+    ----------
+    bst : NumPy array
+        Saturation induction Bs times thickness t of the sample. In Tesla*meter
+    acceleration_voltage : float
+        In Volts
+
+    Returns
+    -------
+    beta : NumPy array
+        Beam deflection in radians
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import diffsims.utils.sim_utils as sim_utils
+    >>> data = np.random.random((100, 100))  # In Tesla*meter
+    >>> acceleration_voltage = 200000  # 200 kV (in Volt)
+    >>> beta = sim_utils.bst_to_beta(data, acceleration_voltage)
+
+    """
+    av = acceleration_voltage
+    wavelength = acceleration_voltage_to_wavelength(av)
+    ##removed defenitions of constants as defined at the top
+    beta = e * wavelength * bst / h
+    return beta
+
+
+def beta_to_bst(beam_deflection, acceleration_voltage):
+    """Calculate Bs * t values from beam deflection (beta).
+
+    Parameters
+    ----------
+    beam_deflection : NumPy array
+        In radians
+    acceleration_voltage : float
+        In Volts
+
+    Returns
+    -------
+    bst : NumPy array
+        In Tesla * meter
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import diffsims.utils.sim_utils as sim_utils
+    >>> data = np.random.random((100, 100))  # In radians
+    >>> acceleration_voltage = 200000  # 200 kV (in Volt)
+    >>> bst = sim_utils.beta_to_bst(data, 200000)
+
+    """
+    wavelength = acceleration_voltage_to_wavelength(acceleration_voltage)
+    beta = beam_deflection
+
+    bst = beta * h / (wavelength * e)
+    return bst
```

Comment:
So this was capitalised because it's the "magnetic field B" as is pretty common in magnetism. Perhaps we should call this magnetic_field or mag_field?